tools/clang/lib/SPIRV/TypeTranslator.cpp

```diff
@@ -135,49 +141,115 @@ bool TypeTranslator::isVectorType(QualType type, QualType *elemType,
   return false;
 }
 
-bool TypeTranslator::is1x1MatrixType(QualType type) {
+bool TypeTranslator::is1x1Matrix(QualType type, QualType *elemType) {
   if (!hlsl::IsHLSLMatType(type))
     return false;
 
   uint32_t rowCount = 0, colCount = 0;
   hlsl::GetHLSLMatRowColCount(type, rowCount, colCount);
 
-  return rowCount == 1 && colCount == 1;
+  const bool is1x1 = rowCount == 1 && colCount == 1;
```

Comment:
Minor: no need for a local variable since it's only used once. Also applies for the following cases.